test/ClangImporter/indirect_fields.swift

```diff
@@ -0,0 +1,19 @@
+// RUN: %target-swift-frontend -typecheck -sdk "" -I %t -I %S/Inputs/custom-modules %s -verify
+
+import IndirectFields
+
+func build_struct(a: Int32, c: Int32, d: Int32) -> StructWithIndirectField {
+    return StructWithIndirectField(__Anonymous_field0: .init(a: a), c: c, d: d)
```

Comment:
This is making me think we're doing this wrong too; we'd really want something like `StructWithIndirectField(a: a, c: c, d: d)`, right? But that seems much harder (and combinatoric, ultimately).